### ISSUE TITLE
Use Mixed-Precision (FP16)

### ISSUE THREAD

**Issue by abondre98** [abondre98/Adaptive-Global-LCA-Advisor-with-Dynamic-EF-Retrieval#65]
○	Enable half-precision training to speed up training and reduce GPU memory usage.
In PyTorch Lightning, you might enable precision=16; in a custom script, consider using Torch AMP:
 scaler = torch.cuda.amp.GradScaler()

with torch.cuda.amp.autocast():
    outputs = lora_model(**batch)
    loss = outputs.loss
scaler.scale(loss).backward()
scaler.step(optimizer)
scaler.update()
